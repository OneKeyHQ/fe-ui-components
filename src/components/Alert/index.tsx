import React, { FC, ReactNode } from 'react';
import cx from 'classnames'

import CheckCircleSolid from '../Icon/react/CheckCircleSolid'
import ExclamationSolid from '../Icon/react/ExclamationSolid';
import InformationCircleSolid from '../Icon/react/InformationCircleSolid';
import XCircleSolid from '../Icon/react/XCircleSolid';
import XSolid from '../Icon/react/XSolid';


export const alertIcons = {
  info: <InformationCircleSolid className="okd-h-5 okd-w-5 okd-text-gray-400" okd-aria-hidden="true" />,
  warning: <ExclamationSolid className="okd-h-5 okd-w-5 okd-text-yellow-400" okd-aria-hidden="true" />,
  error: <XCircleSolid className="okd-h-5 okd-w-5 okd-text-red-400" okd-aria-hidden="true" />,
  success: <CheckCircleSolid className="okd-h-5 okd-w-5 okd-text-green-400" okd-aria-hidden="true" />,
}

export type AlertProps = {
  title: ReactNode;
  content?: ReactNode;
  type?: 'info' | 'warning' | 'error' | 'success';
  closable?: boolean;
  onClose?: () => void;
  className?: string;
}

const defaultProps = {
  type: 'info',
  closable: false,
} as const

const Alert: FC<AlertProps> = ({ type, title, content, closable, onClose, children }) => {
  const alertIconNode = alertIcons[type]
  const textContent = content ?? children

  return (
    <div className={cx("okd-rounded okd-p-4", {
      'okd-bg-gray-50': type === 'info',
      'okd-bg-yellow-50': type === 'warning',
      'okd-bg-red-50': type === 'error',
      'okd-bg-green-50': type === 'success',
    })}>
      <div className="okd-flex">
        <div className="okd-flex-shrink-0">
          {alertIconNode}
        </div>

        <div className="okd-ml-3">
          <h3 className={cx("okd-text-sm okd-font-medium", {
            'okd-text-gray-700': type === 'info',
            'okd-text-yellow-800': type === 'warning',
            'okd-text-red-800': type === 'error',
            "okd-text-green-800": type === 'success',
          })}>
            {title}
          </h3>

          {textContent && (
            <div className={cx("okd-mt-2 okd-text-sm", {
              'okd-text-gray-500': type === 'info',
              'okd-text-yellow-700': type === 'warning',
              'okd-text-red-700': type === 'error',
              "okd-text-green-700": type === 'success'
            })}>
              {textContent}
            </div>
          )}
        </div>

        {closable && (
          <div className="okd-ml-auto okd-pl-3">
            <div className="okd--mx-1.5 okd--my-1.5">
              <button
                type="button"
                className={
                  cx("okd-inline-flex okd-justify-center okd-items-center okd-rounded okd-p-1.5 focus:okd-outline-none focus:okd-ring-2 focus:okd-ring-offset-1", {
                    "okd-text-gray-400 focus:okd-ring-offset-gray-50 focus:okd-ring-gray-600": type === 'info',
                    "okd-text-yellow-500 focus:okd-ring-offset-yellow-50 focus:okd-ring-yellow-600": type === 'warning',
                    "okd-text-red-500 focus:okd-ring-offset-red-50 focus:okd-ring-red-600": type === 'error',
                    "okd-text-green-500 focus:okd-ring-offset-green-50 focus:okd-ring-green-600": type === 'success',
                  })}
                onClick={onClose}
              >
                <span className="okd-sr-only">Dismiss</span>
                <XSolid className="okd-h-5 okd-w-5" okd-aria-hidden="true" />
              </button>
            </div>
          </div>
        )}
      </div>
    </div>
  );
};

Alert.defaultProps = defaultProps;

export default Alert;
