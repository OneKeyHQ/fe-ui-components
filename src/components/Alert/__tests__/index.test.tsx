import React from "react";
import { render, shallow } from "enzyme";
import Alert from "..";

describe("Alert", () => {
  const title = "Testing Title";

  it('should support type of "info" | "warning" | "error" | "success"', () => {
    const info = shallow(<Alert title={title} type="info" />);
    expect(() => info.unmount()).not.toThrow();
    const warning = shallow(<Alert title={title} type="warning" />);
    expect(() => warning.unmount()).not.toThrow();
    const error = shallow(<Alert title={title} type="error" />);
    expect(() => error.unmount()).not.toThrow();
    const success = shallow(<Alert title={title} type="success" />);
    expect(() => success.unmount()).not.toThrow();
  });

  it("should render element", () => {
    const element = render(<Alert title={title} />);
    expect(element).toMatchSnapshot();
  });
});
