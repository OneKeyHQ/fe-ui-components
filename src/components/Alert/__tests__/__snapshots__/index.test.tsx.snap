// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render element 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "okd-rounded okd-p-4 okd-bg-gray-50",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "okd-flex",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "okd-flex-shrink-0",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "okd-h-5 okd-w-5 okd-text-gray-400",
                  "fill": "currentColor",
                  "okd-aria-hidden": "true",
                  "viewBox": "0 0 20 20",
                  "xmlns": "http://www.w3.org/2000/svg",
                },
                "children": Array [
                  Node {
                    "attribs": Object {
                      "clip-rule": "evenodd",
                      "d": "M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a1 1 0 000 2v3a1 1 0 001 1h1a1 1 0 100-2v-3a1 1 0 00-1-1H9z",
                      "fill-rule": "evenodd",
                    },
                    "children": Array [],
                    "name": "path",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "clip-rule": undefined,
                      "d": undefined,
                      "fill-rule": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "clip-rule": undefined,
                      "d": undefined,
                      "fill-rule": undefined,
                    },
                  },
                ],
                "name": "svg",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "fill": undefined,
                  "okd-aria-hidden": undefined,
                  "viewBox": undefined,
                  "xmlns": "http://www.w3.org/2000/xmlns/",
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "fill": undefined,
                  "okd-aria-hidden": undefined,
                  "viewBox": undefined,
                  "xmlns": "",
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {
                "class": "okd-flex okd-flex-col okd-justify-between okd-ml-3",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "class": "okd-text-sm okd-font-medium okd-text-gray-700",
                  },
                  "children": Array [
                    Node {
                      "data": "Testing Title",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "h3",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Node {
            "attribs": Object {
              "class": "okd-flex okd-flex-col okd-justify-between okd-ml-3",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "okd-text-sm okd-font-medium okd-text-gray-700",
                },
                "children": Array [
                  Node {
                    "data": "Testing Title",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "h3",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "class": "okd-flex-shrink-0",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "class": "okd-h-5 okd-w-5 okd-text-gray-400",
                    "fill": "currentColor",
                    "okd-aria-hidden": "true",
                    "viewBox": "0 0 20 20",
                    "xmlns": "http://www.w3.org/2000/svg",
                  },
                  "children": Array [
                    Node {
                      "attribs": Object {
                        "clip-rule": "evenodd",
                        "d": "M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a1 1 0 000 2v3a1 1 0 001 1h1a1 1 0 100-2v-3a1 1 0 00-1-1H9z",
                        "fill-rule": "evenodd",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "clip-rule": undefined,
                        "d": undefined,
                        "fill-rule": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "clip-rule": undefined,
                        "d": undefined,
                        "fill-rule": undefined,
                      },
                    },
                  ],
                  "name": "svg",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "fill": undefined,
                    "okd-aria-hidden": undefined,
                    "viewBox": undefined,
                    "xmlns": "http://www.w3.org/2000/xmlns/",
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "fill": undefined,
                    "okd-aria-hidden": undefined,
                    "viewBox": undefined,
                    "xmlns": "",
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
